core/drivers/crypto/caam/caam_ctrl.c

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2017-2019 NXP
+ *
+ * Brief   CAAM Global Controller.
+ */
+
+/* Global includes */
+#include <initcall.h>
+#include <kernel/panic.h>
+#include <tee_api_types.h>
+
+/* Local includes */
+#include <caam_common.h>
+#include <caam_hal_cfg.h>
+#include <caam_hal_clk.h>
+#include <caam_hal_ctrl.h>
+#ifdef CFG_CRYPTO_HASH_HW
+#include <caam_hash.h>
+#endif
+#include <caam_jr.h>
+#include <caam_pwr.h>
+#include <caam_rng.h>
+#include <caam_utils_mem.h>
+
+/*
+ * Crypto driver initialization function
+ *
+ * Returns:
+ * TEE_SUCCESS              Success
+ * TEE_ERROR_GENERIC        Generic Error (driver init failure)
+ * TEE_ERROR_NOT_SUPPORTED  Driver not supported
```

Comment:
Am I the only one to think this is bad? (documenting the return codes like this)

Reply:
You mean this level of details which easily could become outdated and thus do more harm than good?

Reply:
I think it is useless if not simply wrong.

Reply:
What is usual practice?

Reply:
When using already defined value, prefer centralized their description at one place. This applies to `TEE_ERROR_*` (ref is the GPD TEE spec) as well as your `CAAM_*` status values (one should rather refer to `caam_status.h` for the description).

Here, as returning a `TEE_Result` typed value, the code implicitly expect standard `TEE_Result` values are used, hence I would suggest to simply not describe the return value:
```
/* Crypto driver initialization */
static TEE_Result crypto_driver_init(void)
{
	(...)
}

/* Crypto driver late initialization to complete on-going CAAM operations */
static TEE_Result init_caam_late(void)
{
	(...)
}
```

Reply:
Fine I'll review all functions